src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -0,0 +1,269 @@
+     * @param sampleMap
+     * @return
+     */
+    boolean hasSpecimenType(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("specimenType"))
+                || !EnumUtils.isValidEnumIgnoreCase(SpecimenType.class, sampleMap.get("specimenType"))) {
+            return hasCmoSampleClass(sampleMap);
+        }
+        if (sampleMap.get("specimenType") != "CellLine"
+                || sampleMap.get("specimenType") != "PDX"
+                || sampleMap.get("specimenType") != "Xenograft"
+                || sampleMap.get("specimenType") != "XenograftDerivedCellLine"
+                || sampleMap.get("specimenType") != "Organoid") {
+            return hasCmoSampleClass(sampleMap);
+        }
+        if(sampleMap.get("specimenType") == "exosome"
+                || sampleMap.get("specimenType") == "cfDNA") {
+            return hasSampleOrigin(sampleMap);
+        }
+        return true;        
+    }
+    
+    boolean hasCmoSampleClass(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("cmoSampleClass"))) {
+            return false;
+        }
+        if(!EnumUtils.isValidEnumIgnoreCase(CmoSampleClass.class, sampleMap.get("cmoSampleClass"))) {
+            return false;        
+        }
+        return true;      
+    }
+    
+    boolean hasSampleOrigin(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("sampleOrigin"))) {
+            return false;
+        }
+        if(EnumUtils.isValidEnumIgnoreCase(SampleOrigin.class, sampleMap.get("sampleOrigin"))) {
+            return true;        
+        }
+        return false;
+        
+    }
+    
+    /**
+     * Ensures all variables required to retrieve Nucleid Acid Abbreviation are available
+     * if sampleType is null empty, checks NAtoExtract
+     * if sampleType is "Pooled Library", checks recipe
+     * Else returns true if sampleType has a valid value
+     * @param sampleMap
+     * @return
+     */
+    boolean hasSampleType(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("sampleType"))) {
+            return hasNAtoExtract(sampleMap);
+        }
+        if (sampleMap.get("sampleType") == "Pooled Library") {
+            return hasRecipe(sampleMap);
+        }
+        if(EnumUtils.isValidEnumIgnoreCase(SampleType.class, sampleMap.get("sampleType"))) {
+            return true;        
+        }
+        return false;
+        
+    }
+    
+    boolean hasRecipe(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("recipe"))) {
+            return false;
+        }
+        return true;        
+    }
+    
+    /**
+     * missing this field from sampleManifest, returns true for testing
+     * @param sampleMap
+     * @return
+     */
+    boolean hasNAtoExtract(Map<String, String> sampleMap) {
+        if (Strings.isBlank(sampleMap.get("NAtoExtract"))) {
+            return true;
+        }
+        return true;
+        
```

Comment:
some extra new line spaces to clean up here and below - double check that there aren't any in the above methods that i looked over